### PR TITLE
Set up wifi using credentials from a JSON file

### LINKED ISSUE #404: `particle serial wifi credentials.json` not recognized in newest version 1.28.1
These commands use to work and now the credentials.json is not recognized:
```
$ echo '{"network":"my_ssid","security":"WPA2_AES","password":"my_password"}' > credentials.json
$ particle serial wifi credentials.json
$ particle serial wifi --port /dev/cu.usbmodem141421 credentials.json
```

### PR DESCRIPTION
**Pull request by monkbroc**
Fixes #404

### CODE REVIEW

File: src/cli/serial.js [hunk 1]
```diff
@@ -36,11 +38,30 @@ export default ({ commandProcessor, root }) => {
...
+				description: 'Take the credentials from a JSON file instead of prompting for them'
+			}
+		}, portOption),
 		handler: (args) => {
 			const CloudCommands = require('../cmd/serial');
 			return new CloudCommands(args).configureWifi();
-		}
+		},
+		examples: {
+			'$0 $command': 'Prompt for Wi-Fi credentials and send them to device connected over serial',
```
Comment (stefan-moraru) [inline src/cli/serial.js]: to the device that's*

File: src/cli/serial.js [hunk 2]
```diff
@@ -36,11 +38,30 @@ export default ({ commandProcessor, root }) => {
...
+			'$0 $command --file credentials.json': 'Read Wi-Fi credentials from credentials.json and send them to device connected over serial'
```
Comment (stefan-moraru) [inline src/cli/serial.js]: ^

File: src/cmd/serial.js [hunk 3]
```diff
@@ -966,23 +984,23 @@ class SerialCommand {
 			});
 		});
 
-		const parseSecurityType = (ent, cb) => {
+		const parsesecurity = (ent, cb) => {
 			resetTimeout();
-			if (securityType) {
+			if (opts.security) {
 				let security = 3;
-				if (securityType.indexOf('WPA2') >= 0 && securityType.indexOf('802.1x') >= 0) {
```
Comment (stefan-moraru) [inline src/cmd/serial.js]: Your code looks good and works, this is just a random idea.

Not sure 100% if it works, but it looks kinda cool

```
if (opts.security) {
  const s = opts.security.reduce((acc, current) => { return Object.assign({}, acc, { [current]: true })}, {});

  security = security || (s['WPA2'] && s['802.1x'] && 5);
  isEnterprise = security || isEnterprise;

  security = security || (s['WPA'] && s['802.1x'] && 4);
  isEnterprise = security || isEnterprise;

  security = security || (s['WEP'] && 3);
  isEnterprise = security || isEnterprise;
}
```

Comment (monkbroc) [reply src/cmd/serial.js]: This is not my code. It doesn't have any tests so I'd rather leave it alone.

**Pull request merged**